numpy/linalg/tests/test_linalg.py

```diff
@@ -845,6 +846,111 @@ def test_empty(self):
+                assert_almost_equal(n, 0.0)
+
+            n = norm(at, 0)
+            # Trying to assert equality to `np.dtype(int).type` fails on
+            # 32-bit platforms as it still becomes `np.int64` instead of
+            # `np.int32`. So, this is our workaround.
+            assert_(n.dtype.type in [np.int32, np.int64])
+            assert_almost_equal(n, 2)
+
+            n = norm(at, 1)
+            assert_(issubclass(n.dtype.type, np.inexact))
+            assert_almost_equal(n, 2.0)
+
+            n = norm(at, 2)
+            assert_(issubclass(n.dtype.type, np.inexact))
+            assert_almost_equal(n, 2.0**(1.0/2.0))
+
+            n = norm(at, 4)
+            assert_(issubclass(n.dtype.type, np.inexact))
+            assert_almost_equal(n, 2.0**(1.0/4.0))
+
+            n = norm(at, np.inf)
+            assert_(issubclass(n.dtype.type, np.inexact))
+            assert_almost_equal(n, 1.0)
+
+    def test_matrix_types(self):
+        a = np.array([[1, 0, 1], [0, 1, 1]])
+
+        exact_types = [
+            np.dtype(_).type for _ in np.typecodes['AllInteger']
+        ]
+
+        # float16, float128, and complex256 types are unsupported by `linalg`.
+        # As these types are undefined on some platforms, they need to be
+        # skipped by checking their type codes.
+        inexact_types = [
+            np.dtype(_).type for _ in np.typecodes['AllFloat']
+                                if _ not in ['e', 'g', 'G']
+        ]
+
+        all_types = exact_types + inexact_types
+
+        for each_inexact_types in all_types:
+            at = a.astype(each_inexact_types)
+
+            n = norm(at, -np.inf)
+            assert_(issubclass(n.dtype.type, np.inexact))
+            assert_almost_equal(n, 2.0)
+
+            with warnings.catch_warnings() as w:
+                warnings.simplefilter("ignore", RuntimeWarning)
+                n = norm(at, -1)
+                assert_(issubclass(n.dtype.type, np.inexact))
+                assert_almost_equal(n, 1.0)
+
+            n = norm(at, 1)
+            assert_(issubclass(n.dtype.type, np.inexact))
+            assert_almost_equal(n, 2.0)
+
+            n = norm(at, 2)
+            assert_(issubclass(n.dtype.type, np.inexact))
+            assert_almost_equal(n, 3.0**(1.0/2.0))
+
+            n = norm(at, -2)
+            assert_(issubclass(n.dtype.type, np.inexact))
+            assert_almost_equal(n, 1.0)
+
+            n = norm(at, np.inf)
+            assert_(issubclass(n.dtype.type, np.inexact))
+            assert_almost_equal(n, 2.0)
+
+            n = norm(at, 'fro')
+            assert_(issubclass(n.dtype.type, np.inexact))
+            assert_almost_equal(n, 2.0)
+
+            n = norm(at, 'nuc')
+            assert_(issubclass(n.dtype.type, np.inexact))
+            # Lower bar needed to support low precision floats.
+            # They end up being off by 1 in the 7th place.
+            old_assert_almost_equal(n, 2.7320508075688772, decimal=6)
```

Comment:
I wanted to make sure matrices were tested for types explicitly as it didn't appear there were.